src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TosServiceSpec.scala

```diff
@@ -29,82 +33,137 @@ class TosServiceSpec extends AnyFlatSpec with TestSupport with BeforeAndAfterAll
-    tosServiceEnabledV2.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(false)
-    tosServiceEnabledV2.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
-    tosServiceEnabledV2.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
+    // reject and get ToS status
+    val tosServiceV2 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "0"))
+    val rejectTosStatusResult = tosServiceV2.rejectTosStatus(defaultUser.id, samRequestContext).unsafeRunSync()
+    assert(rejectTosStatusResult, s"rejectTosStatus(${defaultUser.id}) should reject the tos for the user")
+    val userRejectedToS = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatus = tosServiceV2.getTosAdherenceStatus(userRejectedToS).unsafeRunSync()
+    assertResult(expected = false, s"getTosStatus(${defaultUser.id}) should have returned false")(actual = adherenceStatus.userHasAcceptedLatestTos)
   }
 
-  "TosService.isTermsOfServiceStatusAcceptable" should "allow all requests to the API if TOS is disabled" in {
+  "TosService" should "allow users to accept new version of ToS" in {
     assume(databaseEnabled, databaseEnabledClue)
+
     dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
-    val tosServiceDisabled = new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = false))
-    tosServiceDisabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(true)
+
+    // Accept V0
+    val tosServiceV0 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "0"))
+    tosServiceV0.getTosAdherenceStatus(defaultUser).unsafeRunSync() shouldBe TermsOfServiceAdherenceStatus(defaultUser.id, false, false)
+    tosServiceV0.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV0 = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV0 = tosServiceV0.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusV0.userHasAcceptedLatestTos shouldBe true
+
+    // Accept V2
+    val tosServiceV2 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "2"))
+    val adherenceStatusNotAcceptedLatest = tosServiceV2.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusNotAcceptedLatest.userHasAcceptedLatestTos shouldBe false
+    adherenceStatusNotAcceptedLatest.acceptedTosAllowsUsage shouldBe false
+
+    tosServiceV2.acceptTosStatus(userAcceptedV0.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV2 = dirDAO.loadUser(userAcceptedV0.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV2 = tosServiceV2.getTosAdherenceStatus(userAcceptedV2).unsafeRunSync()
+    adherenceStatusV2.userHasAcceptedLatestTos shouldBe true
+    adherenceStatusV2.acceptedTosAllowsUsage shouldBe true
```

Comment:
Again, think about how you can test this with a mocked dao.  If there is logic for rules pertaining to which versions the user can or cannot accept based on the currently accepted version, those rules should live in the TosService and we should have tests for them here.  If this test is just, "A user who previously accepted the ToS can accept a new version of the ToS", then maybe the solution is to just mock dao calls that get the ToS version so that they return a specific version and then run your test trying to accept a new/different version.